sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js

```diff
@@ -222,11 +242,145 @@ Ext.define('Slate.cbl.view.teacher.DashboardController', {
             });
         });
     },
+    
+    onDemonstrationDelete: function(demonstration, operation) {
+        var me = this,
+            dashboardView = me.view,
+            mainGridEl = dashboardView.el.down('.cbl-grid-main'),
+            studentId = demonstration.get('StudentID'),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            skillStudentsTpl = dashboardView.getTpl('skillStudentsTpl'),
+
+            deletedSkills = demonstration.get('Skills'),
+            deletedSkillsLength = deletedSkills.length, deletedSkillIndex = 0, demonstratedSkill,
```

Comment:
no it's the standard practice I described here: https://github.com/SlateFoundation/slate-cbl/pull/10/files#r25317796